pkg/reconciler/revision/revision.go

```diff
@@ -98,18 +98,16 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 
 	var digestGrp errgroup.Group
 	rev.Status.ContainerStatuses = make([]v1.ContainerStatuses, len(rev.Spec.Containers))
+	var failedContainerImage string
 	for i, container := range rev.Spec.Containers {
 		container := container // Standard Go concurrency pattern.
 		i := i
 		digestGrp.Go(func() error {
 			digest, err := c.resolver.Resolve(container.Image,
 				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
 			if err != nil {
-				err = fmt.Errorf("failed to resolve image to digest: %w", err)
-				rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-					v1.RevisionContainerMissingMessage(
-						container.Image, err.Error()))
-				return err
+				failedContainerImage = container.Image
```

Comment:
this still races I think (there's still only one `failedContainerImage` potentially written by multiple goroutines), but I think we could just build and return an error containing the `v1.RevisionContainerMissingMessage` which we can unwrap at the bottom.